### PR TITLE
Parquet: Simplify changes from #3723

### PR DESCRIPTION
**Pull request by rdblue**
This makes a couple of modifications to #3723 to make it more consistent and simpler.
* Instead of creating a fake element type, add "element" to field names directly
* To support direct control of `fieldNames`, move name handling into `ApplyNameMapping`
* Remove unnecessary method, `makeListType`
* Apply similar changes to maps to avoid name mismatch bugs in the future

### CODE REVIEW

**Comment by rdblue** [comment]
@RussellSpitzer can you take a look at this follow-up to #3723?

**Comment by rdblue** [comment]
@SinghAsDev FYI

**Comment by SinghAsDev** [comment]
Thanks @rdblue , this makes sense. I like that we don't have to modify/ construct parquet types with this change.

I have a couple more changes along this, and would like to get your reviews on them. Will likely post them today.

File: parquet/src/main/java/org/apache/iceberg/parquet/ApplyNameMapping.java [hunk 1]
```diff
@@ -97,29 +96,30 @@ public Type primitive(PrimitiveType primitive) {
...
+  }
+
+  public void afterField(Type type) {
+    fieldNames.pop();
+  }
+
   @Override
   public void beforeElementField(Type element) {
-    super.beforeElementField(makeElement(element));
+    // normalize the name to "element" so that the mapping will match structures with alternative names
```
Comment (SinghAsDev) [inline parquet/src/main/java/org/apache/iceberg/parquet/ApplyNameMapping.java:109]: @rdblue would it help if we add some example of this alternate naming source here?  Same for the map's key and value fields.

**Comment by uncleGen** [comment]
#3605 may be fixed by this pr.

**Review by danielcweeks — APPROVED** [review]

**Pull request merged**